.github/workflows/setup-py-creation.yaml

```diff
@@ -0,0 +1,37 @@
+name: create setup.py
+
+on:
+  push:
+    branches:
+    - master
+    paths:
+    - 'poetry.lock'
+    - 'pyproject.toml'
+
+jobs:
+  create-setup-py:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+    - name: Setup python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Generate setup.py
+      run: |
+        python -m pip install --upgrade pip
+        pip install dephell[full]
+        dephell deps convert --from pyproject.toml --to setup.py
+    - name: Fix setup.py format
+      run: |
+        pip install black
+        black setup.py
+    - name: Commit changed setup.py
+      run: |
+        git config --local user.email "renku@datascience.ch"
+        git config --local user.name "RenkuBot"
+        git add setup.py
+        git commit -m 'chore: automatically generate setup.py' || true
+        git push origin ${{ github.head_ref }}
```

Comment:
I believe this isn't going to work because of branch protection - the best you can do is push to a branch and automatically open a PR that could be automatically approved and merged. To do that, you need a token that is not the default token injected by github actions because those by design cannot trigger new actions.

Reply:
You can look at [this workflow](https://github.com/SwissDataScienceCenter/renku/blob/master/.github/workflows/auto-reqs-update-pr.yml) in the main renku repo as an example.

Reply:
I've created a token with under my account which will create PRs with my name. Do we have a RenkuBot token to use here?

Reply:
Wouldn't it be better for the action to run on each branch and update the setup.py as a commit on that branch, if needed. Similar to how we create the `Pipfile.lock` in renku-python

Reply:
Right, so run on the PR branch rather than `master`? That would certainly be easier

Reply:
I thought of it in the beginning but I don't like to have a commit added to my PR/branches automatically.